### ISSUE TITLE
Communication Standard Template

### ISSUE THREAD

**Issue by oeed** [oeed/CraftOS-Standards#45]
A template is required for communication standards. The [existing template](https://github.com/oeed/CraftOS-Standards/blob/master/Standard%20Proposal%20Guidelines.md) needs to be renamed to indicate that it is specific to file formats. There should also be a templates folder, unless there's some way you've integrated these in on `tutorial` @viluon?

**Comment by ghost** [comment]
@oeed ooops I was so dumb, I  wrote one into the rednet pull request, it is in there...

**Comment by ghost** [comment]
oh, this is an issue, not a pull request, I´m not very familiar with Github yet.

**Comment by oeed** [comment]
No problem! Yeah, so for this project, issues are basically TODOs that (basically) anyone can do whereas pull requests are changes that need to be approved.

**Comment by viluon** [comment]
Agreed @oeed, I'll rewrite templates in `tutorial`. I'll post updates about the progress here (will change from `todo` to `work in progress` when I actually start committing).

**Comment by oeed** [comment]
To be honest @viluon I really think we need to get `tutorial` merged ASAP, it's leaving `master` lagging too far behind where we need it.

Can we maybe merge it and instead make smaller branches?

**Comment by viluon** [comment]
I was just thinking about that, let's talk on [Gitter](https://gitter.im/oeed/CraftOS-Standards) @oeed (this is an unrelated Issue).

**Comment by ghost** [comment]
I´ve got an idea for templates, see #46

**Comment by oeed** [comment]
As mentioned in #47, there is quite a difference between _transmission methods_ and _protocols_. Hence, we actually need _two_ different templates.

**Comment by ghost** [comment]
@oeed I think Rednet in a protocol too, but we should add another header Table Row, which contains Information if it can be used as Transmission Method. And Rednet is currently the only  ProtocolAndTrasmissionMethod thing. (once its accepted, I´ll release some Protocols based on Rednet)

**Comment by oeed** [comment]
I might be wrong, but I feel like a protocol is more about _data_ being sent, not the method being sent. But I can see what you're saying.

@demhydraz or @viluon do you have any thoughts on this?

**Comment by demhydraz** [comment]
> As mentioned in #47, there is quite a difference between transmission methods and protocols. Hence, we actually need two different templates.

This is right. Rednet is a transmission method. And something that'd send diffs along for file updating (say, like rsync/ftp) is a protocol.

**Comment by oeed** [comment]
Right, so one could say that protocols are to transmission methods what data formats are to file systems? 

In that case, yes there needs to be two.

**Comment by demhydraz** [comment]
Yeah.

**Comment by ghost** [comment]
Yes, but there are also Transmission Methods which are Protocols. (like a file system in a file)

as writing this I got an Idea:
Why don´t Sort them in Folders like that:

```
=Communication
 =Modem
  -README.md
  -gps.md
  =rednet
   -README.md
   -chat.md
   -dns.md
```

or

```
=Communication
 =Modem
  -gps.md
  -rednet.md
  =rednet
   -chat.md
   -dns.md
```

maybe also without a modem folder

the upper variant will easily show, which Protocol is based on which and also would view TransmissionMethodProtocols as Folder which is easy to understand. The only problem is that I think File names with just big chars should be meta...

EDIT: that would also provide the need for 2 different templates

**Comment by demhydraz** [comment]
> Yes, but there are also Transmission Methods which are Protocols. (like a file system in a file)


ehh?

**Comment by ghost** [comment]
@demhydraz Rednet for example, or maybe just rednet

**Comment by demhydraz** [comment]
You lost me. Either something is a transmission method (LyqydNet, Rednet, just using `modem`...) or it's a protocol (diffsync, tror, chat)

**Comment by oeed** [comment]
@justsomeoneelseagain a protocol defines what the data is you're sending and how to parse it. A transmission method is simple a road for the data to go down.
